openpifpaf/network/losses/composite.py

```diff
@@ -31,7 +31,12 @@ def __init__(self, head_net: heads.CompositeField3):
             + ['{}.{}.scales{}'.format(head_net.meta.dataset, head_net.meta.name, i + 1)
                for i in range(self.n_scales)]
         )
-
+        w = head_net.meta.weights
+        n_fields=head_net.meta.n_fields
+        self.weights = torch.ones([1, n_fields, 1, 1])
```

Comment:
Ok, sorry, I am going a bit back on what I said before. To preserve behavior to previous unweighted case:
Initialize `self.weights = None` and only set it when a w is given. In the forward functions, you then only apply the weight when `self.weights is not None`. Then the behavior is exactly the same as before.